flask/app.py

```diff
@@ -1370,6 +1392,32 @@ def url_defaults(self, f):
         self.url_default_functions.setdefault(None, []).append(f)
         return f
 
+    def _find_error_handler(self, e):
+        """Finds a registered error handler for the request’s blueprint.
+        If neither blueprint nor App has a suitable handler registered, returns None
+        """
+        exc_class, code = self._ensure_exc_class(e)
+        
+        def find_superclass(d):
+            if not d:
+                return None
+            for superclass in exc_class.mro():
+                if superclass is BaseException:
+                    return None
+                handler = d.get(superclass)
+                if handler is not None:
+                    return handler
```

Comment:
>seems kinda arbitrary, but makes things easier for us here.

I suppose otherwise errorhandler and route registration has to be threadsafe.

>i’ll just add d[exc_class] = handler

Yup, sounds good.